gpytorch/lazy/lazy_evaluated_kernel_tensor.py

```diff
@@ -229,11 +229,10 @@ def diag(self):
         x1 = self.x1
         x2 = self.x2
 
-        res = super(Kernel, self.kernel).__call__(x1, x2, diag=True, batch_dims=self.batch_dims, **self.params)
+        res = super().__call__(x1, x2, diag=True, last_dim_is_batch=self.last_dim_is_batch, **self.params)
```

Comment:
this still needs to point to the kernel's method: `super(Kernel, self.kernel)`

Reply:
Should be fixed now